src/appmetrics.cpp

```diff
@@ -388,10 +397,22 @@ static void sendData(const std::string &sourceId, unsigned int size, void *data)
 	memcpy(dataCopy, data, size);
 	payload->data = dataCopy;
 	payload->size = size;
-
-	async->data = payload;
-	uv_async_init(uv_default_loop(), async, (uv_async_cb)emitMessage);
-	uv_async_send(async);
+	payload->next = NULL;
+
+	// Put the next message on the end of the queue.
+	// (So they are sent in the same order we added them.)
+	uv_mutex_lock(messageListMutex);
+	MessageData** tail = &messageList;
+	while( *tail != NULL ) {
+		tail = &((*tail)->next);
+	}
+	(*tail) = payload;
```

Comment:
Could we maintain a tail pointer to prevent the need to traverse the list?
Have we considered having a max limit on the size of the list? We could always drop messages if it gets too big (eg if something is hogging the event loop for a while.